paddle/cinn/common/integer_set.cc

```diff
@@ -0,0 +1,460 @@
+    return true;
+  }
+  std::optional<bool> prove_lt = ProveLT(lhs, rhs);
+  if (prove_lt.has_value() && prove_lt.value()) {
+    return true;
+  }
+  return std::nullopt;
+}
+
+std::optional<bool> SymbolicExprAnalyzer::ProveGE(const ir::Expr& lhs,
+                                                  const ir::Expr& rhs) const {
+  if (lhs == rhs) {
+    return true;
+  }
+  if (lhs == SymbolicExprLimit::positive_inf ||
+      rhs == SymbolicExprLimit::negative_inf) {
+    return true;
+  }
+  if (rhs == SymbolicExprLimit::positive_inf ||
+      lhs == SymbolicExprLimit::negative_inf) {
+    return false;
+  }
+  ir::Expr diff = AutoSimplify(ir::Sub::Make(lhs, rhs), var_intervals_);
+  VLOG(6) << "diff of " << ir::Sub::Make(lhs, rhs) << " = " << diff;
+  if (diff.is_constant() && diff.get_constant() >= 0) {
+    return true;
+  }
+  if (diff.is_constant() && diff.get_constant() < 0) {
+    return false;
+  }
+  ir::Expr diff_lower_bound = LowerBound(diff);
+  VLOG(6) << "lower bound of " << diff << " = " << diff_lower_bound;
+  if (diff_lower_bound.is_constant() && diff_lower_bound.get_constant() >= 0) {
+    return true;
+  }
+  ir::Expr diff_upper_bound = UpperBound(diff);
+  VLOG(6) << "upper bound of " << diff << " = " << diff_upper_bound;
+  if (diff_upper_bound.is_constant() && diff_upper_bound.get_constant() < 0) {
+    return false;
+  }
+  return std::nullopt;
+}
+
+std::optional<bool> SymbolicExprAnalyzer::ProveLE(const ir::Expr& lhs,
+                                                  const ir::Expr& rhs) const {
+  return ProveGE(rhs, lhs);
+}
+
+std::optional<bool> SymbolicExprAnalyzer::ProveGT(const ir::Expr& lhs,
+                                                  const ir::Expr& rhs) const {
+  if (lhs == rhs) {
+    return false;
+  }
+  if (lhs == SymbolicExprLimit::positive_inf ||
+      rhs == SymbolicExprLimit::negative_inf) {
+    return true;
+  }
+  if (rhs == SymbolicExprLimit::positive_inf ||
+      lhs == SymbolicExprLimit::negative_inf) {
+    return false;
+  }
+  ir::Expr diff = AutoSimplify(ir::Sub::Make(lhs, rhs), var_intervals_);
+  VLOG(6) << "diff of " << ir::Sub::Make(lhs, rhs) << " = " << diff;
+  if (diff.is_constant() && diff.get_constant() > 0) {
+    return true;
+  }
+  if (diff.is_constant() && diff.get_constant() <= 0) {
+    return false;
+  }
+  ir::Expr diff_lower_bound = LowerBound(diff);
+  VLOG(6) << "lower bound of " << diff << " = " << diff_lower_bound;
+  if (diff_lower_bound.is_constant() && diff_lower_bound.get_constant() > 0) {
+    return true;
+  }
+  ir::Expr diff_upper_bound = UpperBound(diff);
+  VLOG(6) << "upper bound of " << diff << " = " << diff_upper_bound;
+  if (diff_upper_bound.is_constant() && diff_upper_bound.get_constant() <= 0) {
+    return false;
+  }
+  return std::nullopt;
+}
+
+std::optional<bool> SymbolicExprAnalyzer::ProveLT(const ir::Expr& lhs,
+                                                  const ir::Expr& rhs) const {
+  return ProveGT(rhs, lhs);
```

Comment:
```suggestion
  return !ProveGT(rhs, lhs);
```
这里是写错了？也不能直接取反，这里是optional

Reply:
这里是把参数中的lhs和rhs换了顺序，a < b 跟 b > a是等价的。